src/lib/annotations/doc/DocDrawingDialog.js

```diff
@@ -1,41 +1,254 @@
+     * @return {DocDrawingDialog} Drawing dialog instance
+     */
+    constructor(data) {
+        super(data);
+
+        this.postDrawing = this.postDrawing.bind(this);
+        this.deleteAnnotation = this.deleteAnnotation.bind(this);
+    }
+
     /**
-     * Empty stub to avoid unexpected behavior.
+     * [destructor]
+     *
+     * @return {void}
+     */
+    destroy() {
+        this.unbindDOMListeners();
+        this.removeAllListeners();
+
+        if (!this.element) {
+            return;
+        }
+
+        this.element.removeEventListener('click', annotatorUtil.prevDefAndStopProp);
+        if (this.pageEl && this.pageEl.contains(this.element)) {
+            this.pageEl.removeChild(this.element);
+        }
+    }
+
+    /**
+     * Returns whether or not the dialog is able to be seen
+     *
+     * @public
+     * @return {boolean} Whether or not the dialog is able to be seen
+     */
+    isVisible() {
+        return this.visible;
+    }
+
+    /**
+     * Save the drawing thread upon clicking save. Will cause a soft commit.
      *
      * @override
      * @protected
+     * @return {void}
      */
     addAnnotation() {}
 
     /**
-     * Empty stub to avoid unexpected behavior.
+     * Empty stub to avoid unexpected behavior. Removing a drawing annotation can only be done by deleting the thread.
      *
      * @override
      * @protected
+     * @return {void}
      */
     removeAnnotation() {}
 
     /**
+     * Bind dialog button listeners
+     *
+     * @protected
+     * @return {void}
+     */
+    bindDOMListeners() {
+        if (this.commitButtonEl) {
+            this.commitButtonEl.addEventListener('click', this.postDrawing);
+            this.commitButtonEl.addEventListener('touchend', this.postDrawing);
+        }
+
+        if (this.deleteButtonEl) {
+            this.deleteButtonEl.addEventListener('click', this.deleteAnnotation);
+            this.deleteButtonEl.addEventListener('touchend', this.deleteAnnotation);
+        }
+    }
+
+    /**
+     * Unbind dialog button listeners
+     *
+     * @protected
+     * @return {void}
+     */
+    unbindDOMListeners() {
+        if (this.commitButtonEl) {
+            this.commitButtonEl.removeEventListener('click', this.postDrawing);
+            this.commitButtonEl.removeEventListener('touchend', this.postDrawing);
+        }
+
+        if (this.deleteButtonEl) {
+            this.deleteButtonEl.removeEventListener('click', this.deleteAnnotation);
+            this.deleteButtonEl.removeEventListener('touchend', this.deleteAnnotation);
+        }
+    }
+
+    /**
      * Sets up the drawing dialog element.
      *
+     * @protected
      * @param {Annotation[]} annotations - Annotations to show in the dialog
      * @param {HTMLElement} threadEl - Annotation icon element
      * @return {void}
+     */
+    setup(annotations) {
+        // Create outermost element container
+        this.element = document.createElement('div');
```

Comment:
Maybe refer to this as the wrapper or container so it's a little more specific? e.g. DrawingDialogContainerEl

Reply:
How would this dialog be set up on mobile?

Reply:
It seems to work the same exact way on mobile (unless DOM trees are set up differently).

Reply:
Sorry I meant do we need any mobile specific dialog stuff here. Assuming this will be similar to the plain highlight dialog, which has a differently styled dialogs/functionality on mobile devices

Reply:
Wait just noticed your comment about the design matching on mobile and desktop. This should be fine then.